device-plugin/ccnp-device-plugin/README.md

```diff
@@ -0,0 +1,142 @@
+And will also facilitate other CCNP requirements like mount certain directories to CCNP service PODs and workload PODs.
+
+## Introduction
+
+Currently, the CCNP device plugin has following capabilities:
+- expose tdx guest device node in TDVM to PODs
+- mount Unix Domain Socket dir /run/ccnp/uds into CCNP daemonset PODs and workload PODs to enable intra-node communication
+
+The CCNP device plugin will respond to following resource request defined in POD definition yaml:
+```
+    resources:
+      limits:
+        tdx.intel.com/tdx-guest: 1    
+```
+
+## Installation
+
+### Prerequisite
+The CCNP device plugin need to deploy on VM nodes with guest TEE devices(currently only TDX guest device is supported). So the deployment
+of the plugin daemonset is based on the node label set by [Node Feature Discovery](https://github.com/kubernetes-sigs/node-feature-discovery/).
+So we need to install the NFD and corresponding label rules.
+
+1. setup following udev rule to enable other user in the node to read or write to tdx guest device node
+```
+cat /etc/udev/rules.d/90-tdx.rules
+SUBSYSTEM=="misc",KERNEL=="tdx-guest",MODE="0666"
+
+```
+After adding the rule, you can restart the node or run following command to trigger the update:
+```
+udevadm trigger
+```
+
+2. prepare the shared Unix Domain Socket directory to be mounted to both ccnp service pods and workload pods
+```
+mkdir -p /run/ccnp/uds
+chmod o+w /run/ccnp/uds
+```
+
+3. deploy NFD
+
+> Note: when node-feature-discovery new [release v0.14](https://github.com/kubernetes-sigs/node-feature-discovery/issues/1250) is ready, below command can be used to deploy NFD with TDVM support:
+
+```
+kubectl apply -k https://github.com/kubernetes-sigs/node-feature-discovery/deployment/overlays/default?ref=v0.14
+```
+
+before that, please build own image and deploy:
+```
+git clone https://github.com/kubernetes-sigs/node-feature-discovery.git
+cd node-feature-discovery/
+make image
+kubectl apply -k .
+```
+> Note: please change the imagePullPolicy to 'IfNotPresent' if you build your image locally.
+
+4. deploy NFD label rules
+```
+kubectl apply -f device-plugin/ccnp-device-plugin/deploy/node-feature-rules.yaml
+```
+
+After deployment, following label can be found in the VM node:
+```
+kubectl get node -o json | jq .items[].metadata.labels | grep tdx-guest
+  "intel.feature.node.kubernetes.io/tdx-guest": "enabled",
+```
+Above label can be used as node selector by CCNP device plugin daemonset and CCNP services daemonset.
+
+
+### Build docker image
+The Dockerfile for the service can be found under device-plugin/ccnp-device-plugin/container directory. 
+Use the following command to build the image:
+```
+docker build -t ccnp_device_plugin:0.1 -f device-plugin/ccnp-device-plugin/container/Dockerfile .
+```
+
+> Note: if you are using containerd as the default runtime for kubernetes. Please remember to use the following command to import the image into containerd:
```

Comment:
if you are using containerd as the default runtime for kubernetes, don't forget to use the following commands to import the image into containerd first:

Reply:
updated, thanks!